apps/examples/testcase/le_tc/network/tc_net_poll.c

```diff
@@ -0,0 +1,182 @@
+ */
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <poll.h>
+#include <pthread.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netutils/netlib.h>
+#include <tinyara/os_api_test_drv.h>
+#include "tc_internal.h"
+
+/**
+* @testcase             :tc_net_poll_p
+* @brief                :check for poll in multi-thread env
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_multi_thread_poll_p(int ret, int sd)
+{
+	TC_ASSERT_LEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+	TC_SUCCESS_RESULT();
+}
+
+void *wait_event(void *arg)
+{
+	struct pollfd fds[1];
+	int sd = *(int *)arg;
+	int ret = -1;
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	ret = poll(fds, 1, 1000);
+	tc_net_multi_thread_poll_p(ret, sd);
+
+	return NULL;
+}
+
+void *multi_thread_poll(void *arg)
+{
+	int sd = -1;
+	pthread_t thread1;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sd < 0) {
+		printf("socket fail %s:%d", __FUNCTION__, __LINE__);
+		return NULL;
+	}
+
+	pthread_create(&thread1, NULL, wait_event, (void *)&sd);
+
+	close(sd);
+
+	return NULL;
+}
+
+/**
+* @testcase             :tc_net_poll_p
+* @brief                :check for poll
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_p(void)
+{
+	int ret = -1;
+	int sd = -1;
+	struct pollfd fds[1];
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	ret = poll(fds, 1, 1000);
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+
+	ret = close(sd);
+	TC_ASSERT_GEQ("lwip_poll_tc_close", ret, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+
+/**
+* @testcase             :tc_net_poll_events_n
+* @brief                :test with wrong value of events
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_events_n(void)
+{
+	int ret = -1;
+	int sd = -1;
+	struct pollfd fds[1];
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sd < 0) {
+		printf("socket fail %s:%d", __FUNCTION__, __LINE__);
+		return;
+	}
+
+	fds[0].fd = sd;
+	fds[0].events = 0;
+
+	ret = poll(fds, 1, 1000);
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+
+	close(sd);
+	TC_ASSERT_GEQ("lwip_poll_tc_close", ret, 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             :tc_net_poll_sd_n
+* @brief                :test with wrong value of socket descriptor
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_sd_n(void)
+{
+	int ret = -1;
+	int sd = 1000;
+	struct pollfd fds[1];
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	ret = poll(fds, 1, 1000);
+	TC_ASSERT_LEQ("lwip_poll", ret, -1);
+	TC_SUCCESS_RESULT();
+}
+
+int net_poll_main(void)
+{
+#ifndef CONFIG_DISABLE_POLL
```

Comment:
You should make a dependency with tc config instead of this conditional.
```
config TC_NET_POLL
	bool "lwip_poll() api"
	default n
	depends on !DISABLE_POLL
```

Reply:
done